src/main/java/net/minecraftforge/registries/DataPackRegistriesHooks.java

```diff
@@ -0,0 +1,101 @@
+package net.minecraftforge.registries;
+
+import com.google.common.collect.ImmutableMap;
+import net.minecraft.core.Holder;
+import net.minecraft.core.Registry;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.data.BuiltinRegistries;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraftforge.event.world.AddBuiltinRegistryEvent;
+import net.minecraftforge.fml.ModLoader;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class DataPackRegistriesHooks
+{
+   private static Map<ResourceKey<? extends Registry<?>>, RegistryAccess.RegistryData<?>> REGISTRY_ACCESS_REGISTRIES_COPY;
+
+   /** Modders, DO NOT USE. Internal use only */
+   public static Map<ResourceKey<? extends Registry<?>>, RegistryAccess.RegistryData<?>> addBuiltinRegistriesEvent(ImmutableMap.Builder<ResourceKey<? extends Registry<?>>, RegistryAccess.RegistryData<?>> builder)
+   {
+      REGISTRY_ACCESS_REGISTRIES_COPY = new HashMap<>(builder.build());
+      return REGISTRY_ACCESS_REGISTRIES_COPY;
```

Comment:
This method should return an immutable view instead of the mutable map.

Reply:
I'm not sure what you mean with "immutable view". I need to pass the mutable map, because it changes afterwards. If I would return the immutable map, all changes through the event would be ignored. Could you specify what you mean?

Reply:
A collection "view" is a wrapper instance around an original collection that is updated when the original collection is updated. It is a view because it does not have the same properties / allowed functions. In this case, the immutable view would not allow mutating the map, only reading. Please see `Collections#unmodifiableMap`.